### PR TITLE
Add CORS policy support

### PR DESCRIPTION
**Pull request by binodmx**
## Purpose
This PR resolves https://github.com/wso2/apk/issues/582 by adding CORS policy config for API Policies. A CORS policy can be added by defining `cORSPolicy` property in `APIPolicy` kind. Following is an sample APIPolicy kind which includes a CORS policy.

```yaml
apiVersion: dp.wso2.com/v1alpha1
kind: APIPolicy
metadata:
  name: sample-api-policy
  namespace: apk
spec:
  override:
    cORSPolicy:
      enabled: true
      accessControlAllowCredentials: false
      accessControlAllowOrigins:
      - "*"
      accessControlAllowHeaders:
      - authorization
      accessControlAllowMethods:
      - GET
      - POST
      accessControlExposeHeaders:
      - "*"
  targetRef:
    group: dp.wso2.com
    kind: API
    name: http-bin-api
    namespace: apk
```

The config dump after attaching a CORS policy to an API is as follows.

```json
"envoy.filters.http.cors":{
   "@type":"type.googleapis.com/envoy.extensions.filters.http.cors.v3.CorsPolicy",
   "allow_origin_string_match":[
      {
         "safe_regex":{
            "regex":".*"
         }
      }
   ],
   "allow_methods":"GET, POST",
   "allow_headers":"authorization",
   "expose_headers":"*",
   "allow_credentials":true
}
```

### CODE REVIEW

**Comment by AmaliMatharaarachchi** [comment]
seems like you haven't updated the pr with operator `make manifests` command changes

**Comment by AmaliMatharaarachchi** [comment]
shall we remove these old gateway level configs. https://github.com/wso2/apk/blob/0fb0c55bd025c1e327879b464e7745be18b7e15b/adapter/config/default_config.go#L56

**Comment by binodmx** [comment]
Old gateway level config `globalCors` is removed by this commit https://github.com/wso2/apk/pull/1120/commits/bdf24683848441bdea851759990c9e2e35322e46.

**Review by Amila-Rukshan — APPROVED** [review]

File: test/integration/integration/tests/api-with-cors-policy.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+					},
+					Method: "OPTIONS",
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Host: "",
+						Method: "OPTIONS",
+					},
+				},
+				Response: http.Response{
```
Comment (Amila-Rukshan) [inline test/integration/integration/tests/api-with-cors-policy.go:55]: shall we add another OPTIONS call to check the not allowed response? For example DELETE method for this test

Comment (binodmx) [reply test/integration/integration/tests/api-with-cors-policy.go:57]: Added an OPTIONS call with not allowed origin.

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by CrowleyRajapakse — APPROVED** [review]

**Pull request merged**